src/Native/System.Native/pal_runtimeinformation.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include "pal_types.h"
+
+extern "C" int32_t GetUnixVersion(char* version, int* capacity);
+
+extern "C" int32_t GetUnixArchitecture();
+
+enum 
```

Comment:
scoped enumeration cannot implicitly convert to underlying intergral types, needs static_cast everywhere, where we only intend to use this enum to represent a name for the integrals. unscoped enumeration serves the purpose here with less complexity.